numba/np/random/generator_methods.py

```diff
@@ -819,3 +820,75 @@ def impl(inst, n, p , size=None):
                 out[i] = random_negative_binomial(inst.bit_generator, n, p)
             return out
         return impl
+
+
+@overload_method(types.NumPyRandomGeneratorType, 'noncentral_chisquare')
+def NumPyRandomGeneratorType_noncentral_chisquare(inst, df, nonc, size=None):
+    check_types(df, [types.Float, float], 'df')
+    check_types(nonc, [types.Float, float], 'nonc')
+    if isinstance(size, types.Omitted):
+        size = size.value
+
+    if is_nonelike(size):
+        def impl(inst, df, nonc, size=None):
+            return np.float64(random_noncentral_chisquare(inst.bit_generator,
+                                                          df, nonc))
+        return impl
+    else:
+        check_size(size)
+
+        def impl(inst, df, nonc, size=None):
+            out = np.empty(size, dtype=np.float64)
+            for i in np.ndindex(size):
```

Comment:
This loop is going to be doing a lot of work anyway, but in general, `out` is going to be C ordered by nature of its construction so iterating through `out.ravel()` or `out.flat()` would probably generate a less involved loop nest.

Reply:
Good idea, I think this should be a generalised (i.e. used in rest of existing distributions as well). I'll add it as a task in the `Generator`s meta-issue.

Reply:
Added as a task in #8519